### ISSUE TITLE
Audio not working

### ISSUE THREAD

**Issue by Farfighters** [rebinf/QuranVideoMaker#95]
I just used the new update for audio, and because I was dealing with an old project, it was fine. But when I played the exported version, there is no audio. Also, there is no waveform and if I click on the timeline after adding a new audio, the software crashes. So currently I can't use the new version due to these issues.

**Comment by Farfighters** [comment]
Previous version is working fine though, so must be to do with the new updates to the audio I think

**Comment by rebinf** [comment]
I just created a new project, it seems to be working as expected, do you have a repro project so I can test it with?

**Comment by Farfighters** [comment]
[TestProject100.zip](https://github.com/user-attachments/files/18270293/TestProject100.zip)
Here is the project, but this is happening with every single audio I have, whether it is .wav or .mp3 and no matter the length.

**Comment by rebinf** [comment]
The repro project really helped.
JazakAllahu khayran.

I think #83 is also related to this, so I've closed it, please reopen if not.

**Comment by Farfighters** [comment]
Assalamualaikum wbt. I'm so sorry for not brining this up when you closed this, but I was very busy and only got back onto the software just now. I still have the same issue? I can interact with all the parts of the software, add verses etc, but when it comes to adding the audio, it crashes after placing it in the timeline. I can move around in the timeline before adding audio, but after adding it, and clicking on the audio, timeline or pressing play button, the software crashes?

**Comment by Farfighters** [comment]
I can share another test project if you want, but I'm having the exact same problems as before. Also, on a side note, do previous version files affect the later ones. i.e. should I delete the old zip and extracted files so they don't mess with the new version. I have done this already whenever I get a problem, but just wondering whether I should keep doing it or it actually does nothing...

**Comment by rebinf** [comment]
Wa alayum salam,

No worries, thank you for reporting the issue.
I managed to replicate the issue, so no need for a repro project just now, thank you.

Since QVM currently doesn't have an installer, it's a good idea to delete the old extracted files before extracting the new zip file to the same folder, as the directory structure might change, or files might get left behind that are no longer part of the app.

JazakAllah khayran

**Comment by Farfighters** [comment]
Jazakallah Khair, I have tested it and everything is working now. Most of the audio function is working perfectly now, the only thing that is still not working is when I pull the audio from the right to cut a bit of the audio out, the waveform no longer matches the audio, test it with a long (10min) audio track to really see the difference. The reason this is important is becuase alhamdulillah you have now added the functionality to combine multiple audio tracks in 1 project, but if I want to crop out most of the audio and only keep a specific amount, it is hard to use the waveform. I noticed that it is actually the same difference between waveform and audio always after cropping it, I think just as I grab the side, the entire waveform shifts so slightly, giving the inaccurate waveform. I'm just raising this as this is also part of the audio issues, I'm not sure whether you were intending to fix that though. Jazakhallah khair for all you effort though, May Allah bless you for this!
